packages/jest-circus/src/run.ts

```diff
@@ -42,34 +42,25 @@ const _runTestsForDescribeBlock = async (
     await _callCircusHook({describeBlock, hook});
   }
 
-  // Tests that fail and are retried we run after other tests
-  const retryTimes = parseInt(global[RETRY_TIMES], 10) || 0;
-  const deferredRetryTests = [];
-
-  for (const child of describeBlock.children) {
-    switch (child.type) {
-      case 'describeBlock': {
-        await _runTestsForDescribeBlock(child);
-        break;
-      }
-      case 'test': {
-        const hasErrorsBeforeTestRun = child.errors.length > 0;
-        await _runTest(child);
-
-        if (
-          hasErrorsBeforeTestRun === false &&
-          retryTimes > 0 &&
-          child.errors.length > 0
-        ) {
-          deferredRetryTests.push(child);
+  if (describeBlock.errors.length === 0) {
```

Comment:
Skip running any tests or nested describe blocks if there are any errors in `beforeAll` blocks. This conditional will bypass the `_runTest` hook and not emit the test lifecycle events.

Reply:
Test retry was moved into this block since it should not run if tests are being skipped due to setup failure.